File: implementations/node/sdk/src/contracts/publication/publication-dependency-upload.ts
import type { AiSdkPublicationDependencyPackage } from "./publication-dependency-package.js";
import type { AiSdkPublicationFileUpload } from "./publication-file-upload.js";

export interface AiSdkPublicationDependencyUpload {
  readonly name: string;
  readonly version: string;
  readonly files?: readonly AiSdkPublicationFileUpload[];
  readonly package?: AiSdkPublicationDependencyPackage;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/publication/publication_dependency_upload.py
from __future__ import annotations

from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from .publication_dependency_package import AiSdkPublicationDependencyPackage
from .publication_file_upload import AiSdkPublicationFileUpload


@dataclass(frozen=True)
class AiSdkPublicationDependencyUpload(AiSdkWireModel):
    name: str
    version: str
    files: tuple[AiSdkPublicationFileUpload, ...] = ()
    package: AiSdkPublicationDependencyPackage | None = None
